pylint/lint/pylinter.py

```diff
@@ -691,27 +721,58 @@ def check_single_file_item(self, file: FileItem) -> None:
         with self._astroid_module_checker() as check_astroid_module:
             self._check_file(self.get_ast, check_astroid_module, file)
 
-    def _check_files(
+    def _lint_files(
         self,
-        get_ast: GetAstProtocol,
-        file_descrs: Iterable[FileItem],
+        ast_mapping: dict[FileItem, nodes.Module | None],
+        check_astroid_module: Callable[[nodes.Module], bool | None],
     ) -> None:
-        """Check all files from file_descrs."""
-        with self._astroid_module_checker() as check_astroid_module:
-            for file in file_descrs:
-                try:
-                    self._check_file(get_ast, check_astroid_module, file)
-                except Exception as ex:  # pylint: disable=broad-except
-                    template_path = prepare_crash_report(
-                        ex, file.filepath, self.crash_file_path
-                    )
-                    msg = get_fatal_error_message(file.filepath, template_path)
-                    if isinstance(ex, AstroidError):
-                        self.add_message(
-                            "astroid-error", args=(file.filepath, msg), confidence=HIGH
-                        )
-                    else:
-                        self.add_message("fatal", args=msg, confidence=HIGH)
+        """Lint all AST modules from a mapping.."""
+        for fileitem, module in ast_mapping.items():
+            if module is None:
+                continue
+            try:
+                self._lint_file(fileitem, module, check_astroid_module)
+            except astroid.AstroidError as ex:
+                template_path = prepare_crash_report(
+                    ex, fileitem.filepath, self.crash_file_path
+                )
+                msg = get_fatal_error_message(fileitem.filepath, template_path)
+                self.add_message(
+                    "astroid-error", args=(fileitem.filepath, msg), confidence=HIGH
+                )
```

Comment:
Not sure here, but did we remove the distinction between pylint error and astroid error ? It seems the "fatal" message can't be raised anymore.

Reply:
I think this still exists? It seems to be covered?

Reply:
I made a search for `"fatal"` and found nothing. I think we'd raise an 'astroid-error' now instead of 'fatal' ? Maybe we don't need the distinction and can simplify (the suggestion of new error was to keep that behaviour)

Reply:
`get_fatal_error_message` is just a helper message that returns a string that points to the template. I don't think we need to change anything here.

Reply:
I'm thinking of ``self.add_message("astroid-error", ``, previously there was a ``self.add_message("fatal", `` that is not raised anymore. I.e. The new code always raise an astroid-error, even when the error come from pylint or somewhere else like an OSError "no space in device", or something unexpected.

Reply:
Oh, my bad. I didn't read this correctly. You're right, this is a mistake. Fix incoming.